pkg/applicationserver/applicationserver.go

```diff
@@ -748,8 +753,10 @@ func (as *ApplicationServer) handleUp(ctx context.Context, up *ttnpb.Application
 	}
 	switch p := up.Up.(type) {
 	case *ttnpb.ApplicationUp_JoinAccept:
+		as.lastSeenMap.storeLastSeenFromUplink(ctx, up)
```

Comment:
It's better to do those inside `handle*`, preferably in a `defer` if no error occurred.